File: application/java-credit-card-product-service/src/main/java/cloud/ambar/creditCardProduct/commands/DefineProductCommand.java
package cloud.ambar.creditCardProduct.commands;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Data
@NoArgsConstructor
@AllArgsConstructor
public class DefineProductCommand {
    private String productIdentifierForAggregateIdHash;
    private String name;
    private int interestInBasisPoints;
    private int annualFeeInCents;
    private String paymentCycle;
    private int creditLimitInCents;
    private int maxBalanceTransferAllowedInCents;
    private String reward;
    private String cardBackgroundHex;
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCardEnrollment/Application/Commands/DefineProduct/DefineProductCommand.cs
using MediatR;

namespace CreditCardEnrollment.Application.Commands.DefineProduct;

public record DefineProductCommand(
    string Name,
    int InterestInBasisPoints,
    int AnnualFeeInCents,
    string PaymentCycle,
    int CreditLimitInCents,
    int MaxBalanceTransferAllowedInCents,
    string Reward,
    string CardBackgroundHex) : IRequest<string>;
